### DISCUSSION TITLE
Is it possible to disable Agroal connection pooling?

### DISCUSSION THREAD

**Discussion by nscuro** [quarkusio/quarkus#35041] [Q&A] [answered]
I have multiple services, and potentially multiple instances of those services, all accessing the same PostgreSQL database.

Instead of having each instance maintain their own connection pool, I'm looking at [PgBouncer](https://www.pgbouncer.org/) to handle connection pooling centrally. FWIW, when a central connection pool is used, clients normally use "direct" JDBC connections. In frameworks like Spring, this is usually done by replacing the HikariCP data source with a direct data source. For Quarkus however, it doesn't seem like Agroal can be bypassed.

Of course I checked the docs, but was unable to find any hints towards how Agroal connection pooling can be disabled.

So, can it be disabled? If not, any tips or best practics on how to use Quarkus with PgBouncer (or similar tech)?

[a comment by an automated account is not shown]

**Comment by Sanne** [comment, marked as answer]
Hi, have a look at this property - is that what you're looking for?
 - https://quarkus.io/guides/datasource#quarkus-agroal_quarkus.datasource.jdbc.pooling-enabled

Essentially it's a feature of Agroal - that's good because it means you're still benefiting from the integrations Quarkus/Agroal and all other frameworks, but disabling its internal pool.

**Reply by nscuro** [reply]
That's exactly what I was looking for, thank you! 

Turns out a script blocker extension in my browser prevented the Quarkus docs from loading all available properties. Only noticed this because the deep link you provided did not work at first... Oh man.

**Comment by barreiro** [comment]
Agroal has a 'pool-less' mode for this use case, that can be enabled by [quarkus.datasource.jdbc.pooling-enabled](https://quarkus.io/guides/datasource#quarkus-agroal_quarkus.datasource.jdbc.pooling-enabled).

You would not want to use a default datasource and loose all the features provided by Agroal, like transaction integration, leak detection, metrics, etc. This mode also allows for a few optimizations in the pool.

We also have a native pool (alternative to PgBouncer) called [pgagroal](https://agroal.github.io/pgagroal/) that has some nice features as well. Check it out!

**Reply by nscuro** [reply]
Thanks, I did not know about pgagroal! We'll do some testing with a selection of poolers, and I added pgagroal to said selection. :)

**Comment by naah69** [comment]
@Sanne @barreiro 
Why it still use AgroalDataSource when quarkus.datasource.jdbc.pooling-enabled=false？

<img width="570" alt="image" src="https://github.com/quarkusio/quarkus/assets/25682169/4a9b67f0-3d2f-442a-9fcb-47709d54c68f">

**Reply by yrodiere** [reply]
Because this disables pooling, not Agroal.
